Assets/MixedRealityToolkit-SDK/Features/Audio/Influencers/AudioOccluder.cs

```diff
@@ -0,0 +1,145 @@
+        /// Setting this value to 22000.0 effectively disables the effect.
+        /// </summary>
+        /// <remarks>
+        /// Chaining occluders will result in the lowest of the cutoff frequencies being applied to the sound.
+        /// The CutoffFrequency range is 0.0 - 22000.0 (0 - 22kHz), inclusive.
+        /// The default value is 5000.0 (5kHz).
+        /// </remarks>
+        [Tooltip("Frequency above which sound will not be heard after applying occlusion.")]
+        [Range(10.0f, 22000.0f)]
+        [SerializeField]
+        private float cutoffFrequency = 5000.0f;
+        public float CutoffFrequency
+        {
+            get { return cutoffFrequency; }
+            set
+            {
+                // set cutoffFrequency and enforce the specified range
+                if (value < 10.0f)
+                {
+                    cutoffFrequency = 10.0f;
+                }
+                else if (value > 22000.0f)
+                {
+                    cutoffFrequency = 22000.0f;
+                }
+                else
+                {
+                    cutoffFrequency = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Percentage of the audio source volume that will be heard after applying occlusion.
+        /// </summary>
+        /// <remarks>
+        /// VolumePassThrough is cumulative. It is applied to the current volume of the object at the time
+        /// the effect is applied.
+        /// The VolumePassThrough range is from 0.0 - 1.0 (0-100%), inclusive.
+        /// The default value is 1.0.
+        /// </remarks>
+        [Tooltip("Percentage of the audio source volume that will be heard after applying occlusion.")]
+        [Range(0.0f, 1.0f)]
+        [SerializeField]
+        private float volumePassThrough = 1.0f;
+        public float VolumePassThrough
+        {
+            get { return volumePassThrough; }
+            set
+            {
+                // set cutoffFrequency and enforce the specified range
+                if (value < 10.0f)
+                {
+                    volumePassThrough = 10.0f;
+                }
+                else if (value > 22000.0f)
+                {
+                    volumePassThrough = 22000.0f;
+                }
+                else
+                {
+                    volumePassThrough = value;
+                }
+            }
+        }
+
+        // Update is not used, but is kept so that this component can be enabled/disabled.
+        private void Update() { }
+
+        /// <summary>
+        /// Applies the audio effect.
+        /// </summary>
+        /// <param name="soundEmittingObject">The GameObject on which the effect is to be applied.</param>
+        public void ApplyEffect(GameObject soundEmittingObject)
+        {
+            if (!isActiveAndEnabled)
+            { return; }
+
+            AudioSource audioSource = soundEmittingObject.GetComponent<AudioSource>();
```

Comment:
Why don't we cache these components on the `gameObject` and pass the `AudioSource`, `AudioLowPassFilter`, and any other components through as parameters to `ApplyEffect`?

Reply:
This was the pattern in HTK. i used to pass the audiosource and this ended up being simpler (more encapsulated / less prone to error)

Reply:
I'm not keen on this as `GetComponent` is rather expensive and we're calling this a lot in an Update loop. Perhaps we should change this into an event based system with an `IAudioInfluencerHandler`

Reply:
This happens on human scales of time, this is NOT every frame. The occluder cannot cache the emitter as it may change when the user moves around the scene,

Reply:
> The occluder cannot cache the emitter as it may change when the user moves around the scene,

Sounds like we should be using our event system to handle these kinds of things.